src/http/http.js

```diff
@@ -68,40 +68,45 @@ import {
  *    .catch(err => console.log('An error occurred!', err))
  */
 
-function http (endpoint, { 
-  onSuccess=identity, 
-  onFailure=identity,
-  camelizeResponse=true,
-  before=noop,
-  successDataPath,
-  failureDataPath='errors',
+async function http (endpoint, { 
+  before=noop, 
+  __mock_response, // used for unit testing
   ...options
 }={}) {
-  // Apply config middleware, modifying the options in sequence
-  return composeMiddleware(
+
+  const parsedOptions = await composeMiddleware(
     before,
     setDefaults,
     setAuthHeaders,
     addRootToEndpoint,
     addQueryToEndpoint,
     serializeBody,
     includeCSRFToken,
-    filterFetchOptions,
+    extractFetchOptions,
   )({ endpoint, ...options })
-    // Make the fetch call
-    .then(({ endpoint, ...config }) => fetch(endpoint, config))
+
+  const {
+    onSuccess, 
+    onFailure,
+    camelizeResponse,
+    successDataPath,
+    failureDataPath,
+    endpoint: url,
+    fetchOptions,
+  } = parsedOptions
```

Comment:
Thoughts on destructuring this on line 77?

Reply:
I left the variable to make it potentially more readable but if you don't think it's helpful I could consolidate!

Reply:
Nope, that makes sense to me